Sources/SPMBuildCore/PluginContextSerializer.swift

```diff
@@ -0,0 +1,276 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021-2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Basics
+import Foundation
+import PackageGraph
+import PackageLoading
+import PackageModel
+import TSCBasic
+
+typealias WireInput = HostToPluginMessage.InputContext
+
+/// Creates the serialized input structure for the plugin script based on all
+/// the input information to a plugin.
+internal struct PluginContextSerializer {
```

Comment:
This isn't new code but is pulled out from the DefaultPluginScriptRunner, which now deals only with the mechanics of process launching and message sending.